machine/v1beta1/types_powervsprovider.go

```diff
@@ -0,0 +1,156 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PowerVSMachineProviderConfig is the type that will be embedded in a Machine.Spec.ProviderSpec field
+// for a PowerVS virtual machine. It is used by the PowerVS machine actuator to create a single Machine.
+//
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type PowerVSMachineProviderConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// UserDataSecret contains a local reference to a secret that contains the
+	// UserData to apply to the instance.
+	// +optional
+	UserDataSecret *corev1.LocalObjectReference `json:"userDataSecret,omitempty"`
+
+	// CredentialsSecret is a reference to the secret with IBM Cloud credentials.
+	// +optional
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
```

Comment:
Apologies if I missed some during previous reviews.  We do not re-use these types because we do not control the schemas and re-use in different contexts can result in fields that are not honored when/if they are added, lack of control over validation, and lack of control over doc.  A LocalSecretRef and LocalConfigMap ref are appropriate to add for this package.

Changing already messed up fields may be possible if
1. the client serialization is identical
2. we don't tighten validation on the replacement fields
3. any fields dropped were fully unused
4. coordinate with @sanchezl to do a read/write cycle of the type in question during an upgrade.